src/debug/telemetry/telemetry2_thread_info.c

```diff
@@ -0,0 +1,349 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <stdio.h>
+#include <soc.h>
+#include <adsp_debug_window.h>
+
+#include "telemetry2_slot.h"
+
+LOG_MODULE_REGISTER(thread_info);
+
+/* to adsp backend header, data to be shared with user space client*/
+#define THREAD_INFO_MAX_THREADS 16
+
+struct thread_info {
+	char name[14];
+	uint8_t stack_usage;	/* precentage */
+	uint8_t cpu_usage;	/* precentage */
```

Comment:
percentage

Reply:
Actually this is obsolete. Its currently a U(0,8) fixed point value indicating relative stack and cpu usage.